website/docs/configuration/variables.html.md

```diff
@@ -250,6 +250,32 @@ Terraform will perform the following actions:
 Plan: 1 to add, 0 to change, 0 to destroy.
 ```
 
+In cases where a sensitive variable is used in a nested block, this will result in the whole block being redacted. This is because nested blocks represent sets, and when a member of a set is sensitive, the whole set is considered sensitive:
```

Comment:
I don't quite understand the "nested blocks represent sets" statement here. I know that some blocks can be backed by a set, and others by a list, but I thought a nested block was always eventually a cty object. Am I missing the meaning of this sentence?

Reply:
Nope, that is the meaning of the sentence. It could be that #26379 was overly broad -- my understanding was that for `NestedBlock` vals, the whole val will always end up marked, and thus when the value is unpacked into multiple values, all the values are implicitly marked.

Reply:
I think that's only true for blocks backed by sets (e.g. AWS instance `network_interface`). For example, a Google Compute instance `network_interface` block is backed by a list, and the diff can include some arguments which are sensitive when others are not:

![image](https://user-images.githubusercontent.com/68917/95123240-a2b3b900-071f-11eb-9e3a-31373f39fbf9.png)

Reply:
Thanks for raising this ... Testing this through, you're correct in that this doesn't happen on this nested block, so it's not all nested blocks (from the configuration view):

```
      ~ network_interface {
            name               = "nic0"
          # Warning: this attribute value will be marked as sensitive and will
          # not display in UI output after applying this change
          ~ network            = (sensitive)
            # (3 unchanged attributes hidden)

            # (1 unchanged block hidden)
        }
```

I'm not quite sure of the difference between this and `network_interface` on AWS such that I can communicate it in a user-centric point-of-view 😬 Something about the difference between what the SDK understands to be a NestedBlock and what the user understands to be a nested block?

Reply:
Since the non-set case feels pretty intuitive, and we (probably?) can't do anything about the set behaviour to make it match, maybe the docs here can be focused on the set-backed case.

I'm not sure this is clear, but it's the best I could come up with:

> In some cases where a sensitive variable is used in a nested block, the whole block can be redacted. This happens with resources which can have multiple blocks of the same type, where the values must be unique. This looks like:

Reply:
Thank you so much! I straight up used this, in particular because I like how you found a way to talk about [mathematical] sets without using the word "set"